File: lib/models/appointment.dart
class Appointment {
  final String id;
  final String motherNic;
  final String motherName;
  final String motherEmail;
  final String appointmentType; // 'midwife' or 'doctor'
  final String providerName;
  final String providerId;
  final DateTime appointmentDate;
  final String timeSlot;
  final String status; // 'pending', 'completed', 'cancelled'
  final String? notes;
  final String? additionalProblems;
  final DateTime createdAt;
  final DateTime? updatedAt;

  Appointment({
    required this.id,
    required this.motherNic,
    required this.motherName,
    required this.motherEmail,
    required this.appointmentType,
    required this.providerName,
    required this.providerId,
    required this.appointmentDate,
    required this.timeSlot,
    required this.status,
    this.notes,
    this.additionalProblems,
    required this.createdAt,
    this.updatedAt,
  });

  Map<String, dynamic> toJson() {
    return {
      // Don't send id for new appointments
      if (id.isNotEmpty) 'id': int.tryParse(id) ?? 0,
      'motherNic': motherNic,
      'motherName': motherName,
      'motherEmail': motherEmail,
      'appointmentType': appointmentType
          .toUpperCase(), // Convert to uppercase for backend
      'providerName': providerName,
      'providerId': providerId,
      'appointmentDate': appointmentDate.toIso8601String(),
      'timeSlot': timeSlot,
      'status': status.toUpperCase(), // Convert to uppercase for backend
      'notes': notes,
      'additionalProblems': additionalProblems,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
    };
  }

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['id']?.toString() ?? '',
      motherNic: json['motherNic'] ?? '',
      motherName: json['motherName'] ?? '',
      motherEmail: json['motherEmail'] ?? '',
      appointmentType: (json['appointmentType']?.toString() ?? '')
          .toLowerCase(), // Convert from backend uppercase
      providerName: json['providerName'] ?? '',
      providerId: json['providerId'] ?? '',
      appointmentDate: DateTime.parse(json['appointmentDate']),
      timeSlot: json['timeSlot'] ?? '',
      status: (json['status']?.toString() ?? 'pending')
          .toLowerCase(), // Convert from backend uppercase
      notes: json['notes'],
      additionalProblems: json['additionalProblems'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: json['updatedAt'] != null
          ? DateTime.parse(json['updatedAt'])
          : null,
    );
  }

  Appointment copyWith({
    String? id,
    String? motherNic,
    String? motherName,
    String? motherEmail,
    String? appointmentType,
    String? providerName,
    String? providerId,
    DateTime? appointmentDate,
    String? timeSlot,
    String? status,
    String? notes,
    String? additionalProblems,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Appointment(
      id: id ?? this.id,
      motherNic: motherNic ?? this.motherNic,
      motherName: motherName ?? this.motherName,
      motherEmail: motherEmail ?? this.motherEmail,
      appointmentType: appointmentType ?? this.appointmentType,
      providerName: providerName ?? this.providerName,
      providerId: providerId ?? this.providerId,
      appointmentDate: appointmentDate ?? this.appointmentDate,
      timeSlot: timeSlot ?? this.timeSlot,
      status: status ?? this.status,
      notes: notes ?? this.notes,
      additionalProblems: additionalProblems ?? this.additionalProblems,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: maternalbackend/src/main/java/com/example/maternalcare/model/Appointment.java
package com.example.maternalcare.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "appointments")
public class Appointment {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "mother_nic", nullable = false)
    private String motherNic;
    
    @Column(name = "mother_name", nullable = false)
    private String motherName;
    
    @Column(name = "mother_email", nullable = false)
    private String motherEmail;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "appointment_type", nullable = false)
    private AppointmentType appointmentType;
    
    @Column(name = "provider_name", nullable = false)
    private String providerName;
    
    @Column(name = "provider_id")
    private String providerId;
    
    @Column(name = "appointment_date", nullable = false)
    private LocalDateTime appointmentDate;
    
    @Column(name = "time_slot", nullable = false)
    private String timeSlot;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private AppointmentStatus status;
    
    @Column(name = "notes", columnDefinition = "TEXT")
    private String notes;
    
    @Column(name = "additional_problems", columnDefinition = "TEXT")
    private String additionalProblems;
    
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    // Constructors
    public Appointment() {
        this.createdAt = LocalDateTime.now();
        this.status = AppointmentStatus.PENDING;
    }
    
    public Appointment(String motherNic, String motherName, String motherEmail, 
                      AppointmentType appointmentType, String providerName, 
                      LocalDateTime appointmentDate, String timeSlot) {
        this();
        this.motherNic = motherNic;
        this.motherName = motherName;
        this.motherEmail = motherEmail;
        this.appointmentType = appointmentType;
        this.providerName = providerName;
        this.appointmentDate = appointmentDate;
        this.timeSlot = timeSlot;
    }
    
    // Getters and Setters
    public Long getId() {
        return id;
    }
    
    public void setId(Long id) {
        this.id = id;
    }
    
    public String getMotherNic() {
        return motherNic;
    }
    
    public void setMotherNic(String motherNic) {
        this.motherNic = motherNic;
    }
    
    public String getMotherName() {
        return motherName;
    }
    
    public void setMotherName(String motherName) {
        this.motherName = motherName;
    }
    
    public String getMotherEmail() {
        return motherEmail;
    }
    
    public void setMotherEmail(String motherEmail) {
        this.motherEmail = motherEmail;
    }
    
    public AppointmentType getAppointmentType() {
        return appointmentType;
    }
    
    public void setAppointmentType(AppointmentType appointmentType) {
        this.appointmentType = appointmentType;
    }
    
    public String getProviderName() {
        return providerName;
    }
    
    public void setProviderName(String providerName) {
        this.providerName = providerName;
    }
    
    public String getProviderId() {
        return providerId;
    }
    
    public void setProviderId(String providerId) {
        this.providerId = providerId;
    }
    
    public LocalDateTime getAppointmentDate() {
        return appointmentDate;
    }
    
    public void setAppointmentDate(LocalDateTime appointmentDate) {
        this.appointmentDate = appointmentDate;
    }
    
    public String getTimeSlot() {
        return timeSlot;
    }
    
    public void setTimeSlot(String timeSlot) {
        this.timeSlot = timeSlot;
    }
    
    public AppointmentStatus getStatus() {
        return status;
    }
    
    public void setStatus(AppointmentStatus status) {
        this.status = status;
        this.updatedAt = LocalDateTime.now();
    }
    
    public String getNotes() {
        return notes;
    }
    
    public void setNotes(String notes) {
        this.notes = notes;
        this.updatedAt = LocalDateTime.now();
    }
    
    public String getAdditionalProblems() {
        return additionalProblems;
    }
    
    public void setAdditionalProblems(String additionalProblems) {
        this.additionalProblems = additionalProblems;
    }
    
    public LocalDateTime getCreatedAt() {
        return createdAt;
    }
    
    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
    
    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }
    
    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
    
    // Enums
    public enum AppointmentType {
        DOCTOR, MIDWIFE
    }
    
    public enum AppointmentStatus {
        PENDING, CONFIRMED, COMPLETED, CANCELLED
    }
}
